### PR TITLE
Re add Manufacturer and Supplier Pages

### PR DESCRIPTION
**Pull request by sanctusmob**
Re add Manufacturer and Supplier Pages to Sitemap XML.
For some unknown reason they removed in "Fix all reported bugs #53"

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Re add Manufacturer and Supplier Pages to Sitemap XML
| Type?         | improvement / new feature
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | Already tested

### CODE REVIEW

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Could you please use spaces instead of tabs everywhere?

**Comment by sanctusmob** [comment]
Done

File: gsitemap.php [hunk 1]
```diff
@@ -502,6 +508,132 @@ protected function getCategoryLink(&$link_sitemap, $lang, &$index, &$i, $id_cate
...
+     * @param int    $index           index of the current Google Sitemap file
+     * @param int    $i               count of elements added to sitemap main array
+     * @param int    $id_manufacturer manufacturer object identifier
+     *
+     * @return bool
+     */
+    protected function getManufacturerLink(&$link_sitemap, $lang, &$index, &$i, $id_manufacturer = 0)
+    {
+        $link = new Link();
+        if (method_exists('ShopUrl', 'resetMainDomainCache'))
```
Comment (PierreRambaud) [inline gsitemap.php:525]: Missing encapsulation?

File: gsitemap.php [hunk 2]
```diff
@@ -502,6 +508,132 @@ protected function getCategoryLink(&$link_sitemap, $lang, &$index, &$i, $id_cate
...
+                );
+            if (!$this->addLinkToSitemap(
+                $link_sitemap, array(
+                    'type' => 'manufacturer',
+                    'page' => 'manufacturer',
+                    'lastmod' => $manufacturer->date_upd,
+                    'link' => $url,
+                    'image' => $manifacturer_image
+                ), $lang['iso_code'], $index, $i, $manufacturer_id['id_manufacturer']
+            ))
```
Comment (PierreRambaud) [inline gsitemap.php:569]: Missing encapsulation :thinking:

File: gsitemap.php (hunk 1 above)
Comment (sanctusmob) [reply gsitemap.php]: Fixed

File: gsitemap.php (hunk 2 above)
Comment (sanctusmob) [reply gsitemap.php]: Fixed

**Review by PierreRambaud — CHANGES_REQUESTED** [review]
Cs fixer and phpstan are still not happy

**Comment by sanctusmob** [comment]
Can you check the phpstan error?

Line 605    Parameter #1 $id of class Supplier constructor expects null, int given.

From Supplier class
/**
     * SupplierCore constructor.
     *
     * @param int|null $id
     * @param int|null $idLang
     */
    public function __construct($id = null, $idLang = null)

phpstan should not be complain about that...

**Comment by PierreRambaud** [comment]
> SupplierCore

Not in old Prestashop versions: https://github.com/PrestaShop/PrestaShop/blob/1.7.6.8/classes/Supplier.php#L94

**Comment by sanctusmob** [comment]
and? how to solve this? Obvious constructor can get integer values.

**Comment by matks** [comment]
@sanctusmob phpstan can ignore errors using configuration https://phpstan.org/user-guide/ignoring-errors

File: gsitemap.php [hunk 3]
```diff
@@ -502,6 +508,139 @@ protected function getCategoryLink(&$link_sitemap, $lang, &$index, &$i, $id_cate
...
+                    Context::getContext()->shop->domain . Context::getContext()->shop->physical_uri . Context::getContext()->shop->virtual_uri,
+                ], $image_link
+            ) : $image_link;
+
+            $file_headers = (Configuration::get('GSITEMAP_CHECK_IMAGE_FILE')) ? @get_headers($image_link) : true;
+            $manifacturer_image = [];
+            if ($file_headers[0] != 'HTTP/1.1 404 Not Found' || $file_headers === true) {
+                $manifacturer_image = [
+                    'title_img' => htmlspecialchars(strip_tags($manufacturer->name)),
+                    'caption' => Tools::substr(htmlspecialchars(strip_tags($manufacturer->short_description)), 0, 350), /** @phpstan-ignore-line */
```
Comment (atomiix) [inline gsitemap.php:559]:
```suggestion
                    'caption' => Tools::substr(htmlspecialchars(strip_tags($manufacturer->short_description)), 0, 350), /* @phpstan-ignore-line */
```

**Comment by sanctusmob** [comment]
We are ok?

**Review by atomiix — APPROVED** [review]

**Review by AureRita — CHANGES_REQUESTED** [review]
HI @sanctusmob ,

Can we have an issue and the steps to reproduce to let the team QA test it correctly ?

Thank you :)

**Comment by matks** [comment]
Hello,

It looks like this PR is now stale. I close it as our rules state it "Pull Requests may be closed after 30 days of inactivity following a request for modifications." (https://www.prestashop-project.org/maintainers-guide/reviewing-pull-requests/)

Thank you

**Pull request closed without merging**